staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -486,7 +487,21 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 				if err := s.indexer.Update(d.Object); err != nil {
 					return err
 				}
-				isSync := d.Type == Sync
+
+				isSync := false
+				switch {
+				case d.Type == Sync:
+					// Sync events are only propagated to listeners that requested resync
+					isSync = true
+				case d.Type == Replaced:
+					if accessor, err := meta.Accessor(d.Object); err == nil {
+						if oldAccessor, err := meta.Accessor(old); err == nil {
+							// Replaced events that didn't change resourceVersion are treated as resync events
+							// and only propagated to listeners that requested resync
+							isSync = accessor.GetResourceVersion() == oldAccessor.GetResourceVersion()
```

Comment:
I really didn't want to teach this layer about resource version, but if the impact on existing listeners is as big as I think it is, we may have to

Reply:
What's the alternative? Moving this logic to DeltaFIFO could work, since it also has access to the client's Store (via `knownObjects`), but I suspect that's a concurrency bug waiting to happen (namely, the Store could have an older value, since SharedInformer might still be processing).

Reply:
Yeah, if we're going to filter, this is the place